### PR TITLE
[Feature][PR Update] Wiki snippets autocomplete toggle

### PR DESCRIPTION
**Pull request by mfraezz**
Purpose
=======
Closes https://openscience.atlassian.net/browse/OSF-2096
Updates https://github.com/CenterForOpenScience/osf.io/pull/4367 and https://github.com/CenterForOpenScience/osf.io/pull/4696

Changes
========
* UI updated with checkbox/label (via already-edited third-party library).
* Viewmodel updated KO observable and toggling function.
* CSS edited to force everything into its proper place 
    * without all the `float: left;`s, the toggle was lowered about `5px` and no margin / padding changes had any effect.

Side Effects
=========
None

### CODE REVIEW

**Comment by felliott** [comment]
Actually, none of the `float` additions to wiki-page.css are necessary, and are confusing since all of the `<li>`s are already inline-block. If you need to get the checkbox to center with the other buttons, apply `position: relative; top: -5px;`  to it.

**Comment by felliott** [comment]
Hey @mfraezz,

This looks pretty good to me, I just have the few nitpicky things mentioned above.  Please reassign me when you're done.

Cheers,
@felliott

**Comment by mfraezz** [comment]
Proposed changes with `float`s doesn't seem to work: 

screen shot 2015-12-17 at 11 26 59

In `wiki-page.css`:
```css
 #wmd-button-bar {
-    float:left;
     margin: auto;
     margin-bottom: 10px;
     -webkit-box-flex: 0 0 auto;
@@ -244,7 +243,6 @@
     background-color: Silver;
     display: inline-block;
     list-style: none;
-    float:left;
 }
 .wmd-button {
     width: 20px;
@@ -254,7 +252,6 @@
     display: inline-block;
     list-style: none;
     cursor: pointer;
-    float: left;
 }
 .wmd-button > span {
     background-image: url(/static/public/vendor/pagedown/wmd-buttons.png);
@@ -263,7 +260,6 @@
     width: 20px;
     height: 20px;
     display: inline-block;
-    float: left;
 }
 .wmd-spacer1 {
     left: 50px;
@@ -276,7 +272,6 @@
 }
 .wmd-prompt-background {
     background-color: Black;
-    float: left;
 }
 .wmd-prompt-dialog {
     border: 1px solid #999999;
@@ -286,7 +281,8 @@
     margin-left:5px;
     list-style: none;
     display: inline-block;
-    float: left;
+    position: relative;
+    top: -5px;
 }
 .wiki-single-heading {
```

**Comment by mfraezz** [comment]
Other changes implemented. Back to you @felliott

**Comment by felliott** [comment]
You'll need to add it to the label, not the li.

**Comment by mfraezz** [comment]
Ah, I see. That works, but there is some UI difference -- the styling with `float`s has spacers that connect when the buttons wrap down a line, proposed changes has wider gaps:

`float`:
screen shot 2015-12-17 at 11 45 12

`<label>` changes:
screen shot 2015-12-17 at 11 44 48

Not sure which is preferable.

**Comment by felliott** [comment]
> Ah, I see. That works, but there is some UI difference -- the styling with floats has spacers that connect when the buttons wrap down a line, proposed changes has wider gaps:
> 
> float:
>  <https://cloud.githubusercontent.com/assets/5659262/11875702/f3f94ae2-a4b3-11e5-9c75-862530b9c403.png>
> <label> changes:
>  <https://cloud.githubusercontent.com/assets/5659262/11875714/0d820be8-a4b4-11e5-9db7-a8ef8c73df57.png>
> Not sure which is preferable.
> 
> 

I’d argue the latter, both because I think it’ll look weird if they don’t line up and because I possess a nihilistic and all-encompassing hatred of floats.

**Comment by mfraezz** [comment]
That's fair. I deleted that comment because after a minute of looking at it, I agreed.

**Comment by felliott** [comment]
Love it!  Sending to RTM.

File: website/addons/wiki/static/WikiEditor.js [hunk 1]
```diff
@@ -79,6 +80,11 @@ function ViewModel(url, viewText) {
         }
     });
 
+    self.setAutocom = function(auto) {
```
Comment (sloria) [inline website/addons/wiki/static/WikiEditor.js]: This shouldn't be necessary. `ko.observables` are already setter functions.

**Comment by sloria** [comment]
osf___annoying_autocomplete_wiki

A few product suggestions:

* I'm not sure "Autocomplete" is the correct word for the feature. Words in the text of the wiki do not autocomplete; snippets do. Maybe change the text to "Snippets"?
* Given that the snippets autocomplete is obtrusive, I suggest making snippets disabled by default, at least until we implement the desired behavior described in [the ticket](https://openscience.atlassian.net/browse/OSF-2096). Ideally, the popup menu will only appear after typing a few characters in the current word.

Ping @erinmayhood for feedback.

**Comment by sloria** [comment]
One last suggested change: the "Autocomplete" label is slightly too big; it draws a lot of attention amongst the other toolbar buttons.

Changing the label's element from `span` to a `small`...

Before:

osf___annoying_autocomplete_wiki


After:

osf___annoying_autocomplete_wiki

I can make this change when I merge.

**Pull request merged**